gammapy/analysis/steps.py

```diff
@@ -0,0 +1,498 @@
+        self.analysis.excess_map = excess_map_estimator.run(self.analysis.datasets[0])
+
+
+class FitAnalysisStep(AnalysisStepBase):
+    tag = "fit"
+    requires_datasets = True
+    requires_models = True
+
+    def _run(self):
+        """Fitting reduced datasets to model."""  
+        fit_settings = self.analysis.config.fit
+        for dataset in self.analysis.datasets:
+            if fit_settings.fit_range:
+                energy_min = fit_settings.fit_range.min
+                energy_max = fit_settings.fit_range.max
+                #TODO : add fit range in lon/lat 
+                geom = dataset.counts.geom
+                dataset.mask_fit = geom.energy_mask(energy_min, energy_max)
+    
+        self.analysis.log.info("Fitting datasets.")
+        result = self.analysis.fit.run(datasets=self.analysis.datasets)
+        self.analysis.fit_result = result
+        self.analysis.log.info(self.analysis.fit_result)
+
+
+class FluxPointsAnalysisStep(AnalysisStepBase):
+    tag="flux-points"
+    requires_datasets = True
+    requires_models = True
+    
+    def _run(self):
+        """Calculate flux points for a specific model component."""
+        fp_settings = self.analysis.config.flux_points
+        self.analysis.log.info("Calculating flux points.")
+
+        energy_edges = make_energy_axis(fp_settings.energy).edges
+        flux_point_estimator = FluxPointsEstimator(
+            energy_edges=energy_edges,
+            source=fp_settings.source,
+            fit=self.analysis.fit,
+            **fp_settings.parameters,
+        )
+    
+        fp = flux_point_estimator.run(datasets=self.analysis.datasets)
+    
+        self.analysis.flux_points = FluxPointsDataset(
+            data=fp, models=self.analysis.models[fp_settings.source]
+        )
+        cols = ["e_ref", "dnde", "dnde_ul", "dnde_err", "sqrt_ts"]
+        table = self.analysis.flux_points.data.to_table(sed_type="dnde")
+        self.analysis.log.info("\n{}".format(table[cols]))
+
+class LightCurveAnalysisStep(AnalysisStepBase):
+    tag= "light-curve"
+    requires_datasets = True
+    requires_models = True
+
+    def _run(self):
+        """Calculate light curve for a specific model component."""
+        lc_settings = self.analysis.config.light_curve
+        self.analysis.log.info("Computing light curve.")
+        energy_edges = make_energy_axis(lc_settings.energy_edges).edges
+
+        if (
+            lc_settings.time_intervals.start is None
+            or lc_settings.time_intervals.stop is None
+        ):
+            self.analysis.log.info(
+                "Time intervals not defined. Extract light curve on datasets GTIs."
+            )
+            time_intervals = None
+        else:
+            time_intervals = [
+                (t1, t2)
+                for t1, t2 in zip(
+                    lc_settings.time_intervals.start, lc_settings.time_intervals.stop
+                )
+            ]
+
+        self.analysis.check_datasets()
```

Comment:
why is it needed explicitly here?

Reply:
removed it for now but I will reintroduce a similar system to check if the data required for the each step are well defined